File: game-client/src/api/PlayerService.ts
import { Flux } from "reactor-core-js/flux";
import { Single } from "rsocket-flowable";
import { Player, Location } from "../game-idl";

export default interface PlayerService {

    locate(locationStream: Flux<Location.AsObject>): Single<void>

    players(): Flux<Player.AsObject>;
}

File: game-server/src/main/java/org/coinen/reactive/pacman/service/PlayerService.java
package org.coinen.reactive.pacman.service;

import org.coinen.pacman.Location;
import org.coinen.pacman.Player;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface PlayerService {

    Mono<Player> createPlayer(String nickname);

    Mono<Void> disconnectPlayer();

    Mono<Void> locate(Flux<Location> locationStream);

    Flux<Player> players();
}
